src/lib.js

```diff
@@ -677,6 +676,65 @@ var program,
       }
     },
 
+    /**
+     * Identifies whether the input data string is a graphql query or not
+     *
+     * @param {string} dataString - Input data string to check if it is a graphql query
+     * @param {string} contentType - Content type header value
+     * @returns {Object} - { result: true, graphql: {Object} } if dataString is a graphql query else { result: false }
+    */
+    identifyGraphqlRequest: function (dataString, contentType) {
+      try {
+        const rawDataObj = _.attempt(JSON.parse, dataString.replace(/[\r\n]/g, ''));
```

Comment:
@aman-v-singh This regex will remove all individual `\n` and `\r` from data. If we have problem with only `\r\n` than let's only replace that with `\n`.

Reply:
https://stackoverflow.com/questions/42068/how-do-i-handle-newlines-in-json?noredirect=1&lq=1

Reply:
@VShingala I have added a function to handle escape characters.